homeassistant/components/meteo_france/sensor.py

```diff
@@ -98,15 +113,24 @@ def state(self):
         """Return the state."""
         path = SENSOR_TYPES[self._type][ENTITY_API_DATA_PATH].split(":")
         data = getattr(self.coordinator.data, path[0])
-
-        if path[0] == "probability_forecast":
-            # TODO: return often 'null' with France cities. Need investigation
-            # TODO: "probability_forecast" not always available.
-            data = data[0]
-
-        if len(path) == 3:
-            value = data[path[1]][path[2]]
-        value = data[path[1]]
+        try:
+            # Specific case for probability forecast
+            if path[0] == "probability_forecast":
+                if len(path) == 3:
+                    # This is a fix compared to other entitty as first index is always null in API result for unknown reason
+                    value = _find_first_probability_forecast_not_null(data, path)
+                else:
+                    value = data[0][path[1]]
+
+            # General case
+            else:
+                if len(path) == 3:
+                    value = data[path[1]][path[2]]
+                else:
+                    value = data[path[1]]
+        except IndexError:
+            _LOGGER.warning("Expected data is missing in API results for %s", self.name)
+            value = None
```

Comment:
As discussed, I think we should remove the check if this should not happen.

Reply:
OK